File: AuraFit-BackEnd/src/main/java/app/aurafitbackend/DTOS/DisplayDTOS/ProductVariantDTO.java
package app.aurafitbackend.DTOS.DisplayDTOS;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;
import java.util.Set;
import java.util.UUID;

import app.aurafitbackend.Beans.*;
import lombok.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ProductVariantDTO {
    private Long id;
    private String sku;
    private Size size;
    private Color color;
    private Material materials;
    private BigDecimal basePrice;
    private BigDecimal salePrice;
    private Integer stockQuantity;
    private boolean onSale;
    private ProductDTO product;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private List<ProductVariantImage> images;
}

File: aurafit-frontend/src/Models/DTOS/ProductVariantDTO.ts
import {ProductDTO} from "./ProductDTO.ts";
import {Material} from "../Material.ts";
import {Color} from "../Color.ts";
import {Size} from "../Size.ts";
import {ProductVariantImage} from "../ProductVariantImage.ts";


export class ProductVariantDTO {
    id: number
    sku: string
    stockQuantity: number
    size: Size
    color: Color
    material: Material;
    basePrice: number
    salePrice: number
    onSale: boolean
    product: ProductDTO
    createdAt: Date
    updatedAt: Date
    productImages?: ProductVariantImage[];

    constructor(id: number, sku: string, stockQuantity: number, size:
    Size, color: Color, material: Material, basePrice: number,
                salePrice: number, onSale: boolean, product: ProductDTO, createdAt: Date, updatedAt: Date, productImages: ProductVariantImage[]) {
        this.id = id;
        this.sku = sku;
        this.stockQuantity = stockQuantity;
        this.size = size;
        this.onSale = onSale;
        this.color = color;
        this.material = material;
        this.basePrice = basePrice;
        this.salePrice = salePrice;
        this.product = product;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
        this.productImages = productImages;
    }
}